Update JWS API to include public key by default and throw exceptions

bplatz opened the pull request:
## Summary
- Change `create-jws` to include public key in header by default for better usability
- Update `verify-jws` to throw exceptions instead of returning them, matching the previous API behavior
- Fix crypto-js wrapper to properly handle key pair objects

## API Changes

### create-jws
Now supports three calling patterns:
1. `(create-jws payload signing-key)` - includes pubkey in header by default
2. `(create-jws payload signing-key-map)` - uses pubkey from signing-key map
3. `(create-jws payload signing-key {:include-pubkey false})` - explicitly opt-out of including pubkey

### verify-jws  
- Now throws exceptions on verification failure instead of returning Exception objects
- This matches the original behavior and fixes compatibility issues

## Implementation Details
- Simplified implementation using `(not (false? (:include-pubkey opts)))` check
- Updated all tests to work with new API behavior
- Applied cljfmt formatting fixes throughout

## Test Plan
- [x] All existing tests pass
- [x] Tested three calling patterns work correctly
- [x] Verified exception throwing behavior
- [x] CommonJS build tested with all patterns

bplatz committed:
Update JWS API to include public key by default and throw exceptions

- Change create-jws to include public key in header by default
- Support three calling patterns:
  1. (create-jws payload signing-key) - includes pubkey in header
  2. (create-jws payload signing-key-map) - uses pubkey from map
  3. (create-jws payload signing-key {:include-pubkey false}) - opt-out
- Update verify-jws to throw exceptions instead of returning them
- Fix crypto-js wrapper to handle key pair objects correctly
- Update all tests to work with new API behavior
- Simplify implementation using (not (false? ...)) check instead of extra opts handling
- Apply cljfmt formatting fixes

dpetran commented on src/fluree/crypto/jws.cljc:
```diff
@@ -63,52 +63,53 @@
+                       (assoc :kid (:kid opts))
+
+                      ;; Add full public key as JWK if requested  
+                       (and (not (false? (:include-pubkey opts))) public-key)
```
I think this logic would be easier to understand if we just set the default in a top-level `opts` destructuring:
```
[payload signing-key {:keys [account-id include-pubkey? jwk kid] :or {include-pubkey? true} :as opts}]
```
Then we don't have to do repeated lookups on the `opts` map and the can just be `(and include-pubkey? public-key)`.

dpetran replied:
I also think it would be nice to change `:include-pubkey` to `:include-pubkey?` so users can more easily infer that it is a boolean value.

dpetran commented on src/fluree/crypto.cljc:
```diff
@@ -234,50 +234,53 @@
-   (jws/verify jws public-key)))
\ No newline at end of file
+   (let [result (jws/verify jws public-key)]
+     (if (instance? #?(:clj Exception :cljs js/Error) result)
```
Instead of catching and returning an error value in `jws/verify`, can we just throw from there? This seems redundant.

bplatz committed:
Address PR feedback: improve JWS API implementation

- Use destructuring with default value for include-pubkey? parameter
- Rename :include-pubkey to :include-pubkey? to indicate boolean nature
- Remove redundant error handling by throwing directly from jws/verify
- Combine nested let statements into single let for cleaner code
- Update all tests and examples to use new parameter name
- All tests pass with new implementation

bplatz committed:
Remove obsolete SJCL externs references from shadow-cljs config

- Remove sjcl-externs.js from :externs in all build configurations
- Eliminates compilation warnings about missing SJCL extern files
- SJCL was replaced with Noble crypto libraries

bplatz commented:
@dpetran I think that latest push addressed your feedback.

dpetran reviewed (approved):
:football:
